KappaAnalysis/src/Producers/MuonCorrectionsProducer.cc

```diff
@@ -151,26 +133,30 @@ void MuonCorrectionsProducer::Produce(KappaEvent const& event, KappaProduct& pro
 			float scaledPy = muon->get()->p4.Py() * scaleFactor;
 			float scaledPz = muon->get()->p4.Pz() * scaleFactor;
 			float scaledE = TMath::Sqrt(TMath::Power(scaledPx,2) + TMath::Power(scaledPy,2) + TMath::Power(scaledPz,2) + TMath::Power(muonMass,2));
-
+			LOG(DEBUG) << "Corrected: " << "scaledPx: " << scaledPx << " scaledPy: " << scaledPy << " scaledPz: " 
+				<< scaledPz << " scaledE: " << scaledE << " Pt: " << muon->get()->p4.Pt() * scaleFactor;
 			muon->get()->p4.SetPxPyPzE(scaledPx, scaledPy, scaledPz, scaledE);
-		}
-		else if (muonEnergyCorrection != MuonEnergyCorrection::NONE)
-		{
+		} else if (muonEnergyCorrection != MuonEnergyCorrection::NONE) {
 			LOG(FATAL) << "Muon energy correction of type " << Utility::ToUnderlyingValue(muonEnergyCorrection) << " not yet implemented!";
 		}
 
-		if (!settings.GetCorrectOnlyRealMuons() || (settings.GetCorrectOnlyRealMuons() && isRealMuon))
+		if (!settings.GetCorrectOnlyRealMuons() || (settings.GetCorrectOnlyRealMuons() && isRealMuon)) {
 			AdditionalCorrections(muon->get(), event, product, settings);
+			LOG(DEBUG) << "Additional Corrections applied! Please verify that this is correct ...\n";
+		}
+			
 
 		// make sure to also save the corrected lepton and the matched genParticle in the map
 		// if we match genParticles to all leptons
 		if (settings.GetRecoMuonMatchingGenParticleMatchAllMuons())
 		{
+			//TODO: add debug log later
 			product.m_genParticleMatchedMuons[muon->get()] =  &(*product.m_genParticleMatchedMuons[static_cast<KMuon*>(const_cast<KLepton*>(product.m_originalLeptons[muon->get()]))]);
 			product.m_genParticleMatchedLeptons[muon->get()] = &(*product.m_genParticleMatchedLeptons[const_cast<KLepton*>(product.m_originalLeptons[muon->get()])]);
 		}
 		if (settings.GetMatchAllMuonsGenTau())
 		{
+			//TODO: add debug log later
```

Comment:
This also needs to be addressed before merging.

Reply:
See comment above.